module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -688,6 +716,58 @@ namespace module::input {
                             }
                         }
                         else {
+                            /************************************************
+                             *       Torso CoM Position in World (rMWw)     *
+                             ************************************************/
+                            bool update_done = false;
+
+                            for (const auto& side : {BodySide::LEFT, BodySide::RIGHT}) {
+                                const bool& foot_down      = sensors->feet[side].down;
+                                const bool& prev_foot_down = previous_foot_down[side];
+
+                                // Get the Foot to Torso transform for this foot
+                                const Eigen::Affine3d Htf(sensors->Htx[side == BodySide::LEFT ? ServoID::L_ANKLE_ROLL
+                                                                                              : ServoID::R_ANKLE_ROLL]);
+
+                                // Calculate our current Foot to CoM vector for this foot
+                                const Eigen::Vector3d current_rMFt =
+                                    Htf.translation().head<3>() + sensors->rMTt.head<3>();
```

Comment:
Pretty sure that the translation is already 3-dimensional
```suggestion
                                    Htf.translation() + sensors->rMTt.head<3>();
```